web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -1,16 +1,66 @@
 const uriParams = document.location.search.split('?')[1].split('&');
-const comicName = decodeURI(uriParams[0].split('=')[1]);
-const chapter = uriParams[1].split('=')[1];
-const page = uriParams[2].split('=')[1];
-console.log(`BmcSideBar: comic ${comicName}, chapter=${chapter}, page=${page}`);
+const hostOrigin = decodeURIComponent(uriParams[0].split('=')[1]);
+const readerName = decodeURIComponent(uriParams[1].split('=')[1]);
+const comicName = decodeURIComponent(uriParams[2].split('=')[1]);
+const chapter = uriParams[3].split('=')[1];
+const page = uriParams[4].split('=')[1];
+console.log(`BmcSideBar: origin ${hostOrigin} : comic ${comicName}, chapter=${chapter}, page=${page}`);
+
+console.log('Loading Engine');
+const bmcEngine = new BmcEngine(hostOrigin, readerName, comicName, chapter, page);
 
 function BmcMangaList() {
     this._node = document.getElementById('manga-list');
+    this._mode = BmcMangaList.MODE_BROWSE;
+}
+
+BmcMangaList.prototype.MODE_REGISTER = 'register';
+BmcMangaList.prototype.MODE_BROWSE = 'browse';
+
+BmcMangaList.prototype.onAliasClick = function(ev) {
+    const comicLabel = ev.target;
+    console.log(`BmcSideBar: BmcMangaList: onAlias: Label=${ev.target.innerText} id=${ev.target.id} reader=${readerName} name=${comicName}`);
+    const handleAliasError = err => {
+        bmcEngine.removeEventListener(bmcEngine.events.alias.error, handleAliasError);
+        bmcEngine.removeEventListener(bmcEngine.events.alias.complete, handleAliasSuccess);
+        this.setMode(this.MODE_BROWSE);
+        showHideSidePanel();
+        notifyResult('Alias Comic', new Error('Could not alias current page to comicId ' + comicId + ': ' + err.message));
+    }
```

Comment:
You never put ';' at the end of the declaration of a closure?

Reply:
Mistake; I will add the semicolons :)